core/platform/cloud_translate/cloud_translate_emulator.py

```diff
@@ -0,0 +1,107 @@
+# coding: utf-8
+#
+# Copyright 2021 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""An emulator that mocks the core.platform.cloud_translate API. This emulator
+models the Cloud Translate API.
+"""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals # pylint: disable=import-only-modules
+
+import python_utils
+import utils
+
+
+class CloudTranslateEmulator(python_utils.OBJECT):
+    """The emulator mocks the translate_text function from the Cloud Translate
+    API. This emulator can be used in backend testing, or a local dev
+    environment without access to the Cloud Translate API. Expected responses
+    must be passed in before using this emulator for testing. See
+    PREGENERATED_TRANSLATIONS above for some prepopulated responses.
+    """
+
+    def __init__(self):
+        """Initializes the emulator with no expected responses."""
+        self.DEFAULT_RESPONSE = (
+            'Default translation for emulator mode. (See core/platform/'
+            'cloud_translate/cloud_translate_emulator.py for details)')
+
+        # Pre-generated translations for the following phrases:
+        # ('hello world', 'CONTINUE', 'Please continue.', 'Correct!')
+        self.PREGENERATED_TRANSLATIONS = {
+            (u'en', u'pt', u'hello world'): u'Ol\xe1 Mundo',
+            (u'en', u'pt', u'CONTINUE'): u'PROSSEGUIR',
+            (u'en', u'es', u'Please continue.'): u'Por favor continua.',
+            (u'en', u'fr', u'CONTINUE'): u'CONTINUEZ',
+            (u'en', u'fr', u'Please continue.'):
+                u'Continuez s&#39;il vous pla\xeet.',
+            (u'en', u'es', u'CONTINUE'): u'SEGUIR',
+            (u'en', u'zh', u'hello world'): u'\u4f60\u597d\u4e16\u754c',
+            (u'en', u'es', u'Correct!'): u'\xa1Correcto!',
+            (u'en', u'zh', u'Correct!'): u'\u6b63\u786e\u7684\uff01',
+            (u'en', u'zh', u'CONTINUE'): u'\u7ee7\u7eed',
+            (u'en', u'zh', u'Please continue.'): u'\u8bf7\u7ee7\u7eed\u3002',
+            (u'en', u'fr', u'Correct!'): u'Correct!',
+            (u'en', u'pt', u'Correct!'): u'Correto!',
+            (u'en', u'es', u'hello world'): u'Hola Mundo',
+            (u'en', u'pt', u'Please continue.'): u'Por favor continue.',
+            (u'en', u'fr', u'hello world'): u'Bonjour le monde'
+            }
+        self.expected_responses = self.PREGENERATED_TRANSLATIONS
```

Comment:
Please move class constants into the class body:

```suggestion
    DEFAULT_RESPONSE = (
        'Default translation for emulator mode. (See core/platform/'
        'cloud_translate/cloud_translate_emulator.py for details)')

    # Pre-generated translations for the following phrases:
    # ('hello world', 'CONTINUE', 'Please continue.', 'Correct!')
    PREGENERATED_TRANSLATIONS = {
        (u'en', u'pt', u'hello world'): u'Ol\xe1 Mundo',
        (u'en', u'pt', u'CONTINUE'): u'PROSSEGUIR',
        (u'en', u'es', u'Please continue.'): u'Por favor continua.',
        (u'en', u'fr', u'CONTINUE'): u'CONTINUEZ',
        (u'en', u'fr', u'Please continue.'):
            u'Continuez s&#39;il vous pla\xeet.',
        (u'en', u'es', u'CONTINUE'): u'SEGUIR',
        (u'en', u'zh', u'hello world'): u'\u4f60\u597d\u4e16\u754c',
        (u'en', u'es', u'Correct!'): u'\xa1Correcto!',
        (u'en', u'zh', u'Correct!'): u'\u6b63\u786e\u7684\uff01',
        (u'en', u'zh', u'CONTINUE'): u'\u7ee7\u7eed',
        (u'en', u'zh', u'Please continue.'): u'\u8bf7\u7ee7\u7eed\u3002',
        (u'en', u'fr', u'Correct!'): u'Correct!',
        (u'en', u'pt', u'Correct!'): u'Correto!',
        (u'en', u'es', u'hello world'): u'Hola Mundo',
        (u'en', u'pt', u'Please continue.'): u'Por favor continue.',
        (u'en', u'fr', u'hello world'): u'Bonjour le monde'
    }

     def __init__(self):
         """Initializes the emulator with no expected responses."""
         self.expected_responses = self.PREGENERATED_TRANSLATIONS
```

Reply:
Thanks for the suggestion! Done.